frontend/public/components/dashboard/inventory-card/inventory-item.tsx

```diff
@@ -45,102 +54,161 @@ const getStatusGroupIcons = (flags: FlagsObject) => {
 };
 
 export const InventoryItem: React.FC<InventoryItemProps> = React.memo(
-  ({ isLoading, singularTitle, pluralTitle, count, children, error = false }) => {
-    const title = count !== 1 ? pluralTitle : singularTitle;
-    let status: React.ReactNode;
-    if (error) {
-      status = <div className="co-dashboard-text--small text-secondary">Unavailable</div>;
-    } else if (isLoading) {
-      status = <LoadingInline />;
-    } else {
-      status = children;
-    }
-    return (
-      <div className="co-inventory-card__item">
-        <div className="co-inventory-card__item-title">{isLoading || error ? title : `${count} ${title}`}</div>
-        <div className="co-inventory-card__item-status">{status}</div>
+  ({ isLoading, title, count, children, error = false, TitleComponent, ExpandedComponent }) => {
+    const [expanded, setExpanded] = React.useState(false);
+    const pluralizedTitle = count !== 1 ? `${title}s` : title;
+    const titleMessage = isLoading || error ? pluralizedTitle : `${count} ${pluralizedTitle}`;
```

Comment:
Nit: multi-part ternary condition should be enclosed in braces:

```ts
(isLoading || error) ? foo : bar
```

Reply:
fixed